pkg/sdk/schemas.go

```diff
@@ -13,3 +50,324 @@ func (v *Schema) ID() SchemaIdentifier {
+	OrReplace                  *bool            `ddl:"keyword" sql:"OR REPLACE"`
+	Transient                  *bool            `ddl:"keyword" sql:"TRANSIENT"`
+	schema                     bool             `ddl:"static" sql:"SCHEMA"` //lint:ignore U1000 This is used in the ddl tag
+	IfNotExists                *bool            `ddl:"keyword" sql:"IF NOT EXISTS"`
+	name                       SchemaIdentifier `ddl:"identifier"` //lint:ignore U1000 This is used in the ddl tag
+	Clone                      *Clone           `ddl:"-"`
+	WithManagedAccess          *bool            `ddl:"keyword" sql:"WITH MANAGED ACCESS"`
+	DataRetentionTimeInDays    *int             `ddl:"parameter" sql:"DATA_RETENTION_TIME_IN_DAYS"`
+	MaxDataExtensionTimeInDays *int             `ddl:"parameter" sql:"MAX_DATA_EXTENSION_TIME_IN_DAYS"`
+	DefaultDDLCollation        *string          `ddl:"parameter,single_quotes" sql:"DEFAULT_DDL_COLLATION"`
+	Tag                        []TagAssociation `ddl:"keyword,parentheses" sql:"TAG"`
+	Comment                    *string          `ddl:"parameter,single_quotes" sql:"COMMENT"`
+}
+
+func (opts *CreateSchemaOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return ErrInvalidObjectIdentifier
+	}
+	if valueSet(opts.Clone) {
+		if err := opts.Clone.validate(); err != nil {
+			return err
+		}
+	}
+	if everyValueSet(opts.OrReplace, opts.IfNotExists) {
+		return errors.New("IF NOT EXISTS and OR REPLACE are incompatible.")
+	}
+	return nil
+}
+
+func (v *schemas) Create(ctx context.Context, id SchemaIdentifier, opts *CreateSchemaOptions) error {
+	if opts == nil {
+		opts = &CreateSchemaOptions{}
+	}
+	opts.name = id
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type AlterSchemaOptions struct {
+	alter                bool             `ddl:"static" sql:"ALTER"`  //lint:ignore U1000 This is used in the ddl tag
+	schema               bool             `ddl:"static" sql:"SCHEMA"` //lint:ignore U1000 This is used in the ddl tag
+	IfExists             *bool            `ddl:"keyword" sql:"IF EXISTS"`
+	name                 SchemaIdentifier `ddl:"identifier"`
+	NewName              SchemaIdentifier `ddl:"identifier" sql:"RENAME TO"`
+	SwapWith             SchemaIdentifier `ddl:"identifier" sql:"SWAP WITH"`
+	Set                  *SchemaSet       `ddl:"list,no_parentheses" sql:"SET"`
+	Unset                *SchemaUnset     `ddl:"list,no_parentheses" sql:"UNSET"`
+	EnableManagedAccess  *bool            `ddl:"keyword" sql:"ENABLE MANAGED ACCESS"`
+	DisableMangaedAccess *bool            `ddl:"keyword" sql:"DISABLE MANAGED ACCESS"`
+}
+
+func (opts *AlterSchemaOptions) validate() error {
+	var errs []error
+	if !validObjectidentifier(opts.name) {
+		errs = append(errs, ErrInvalidObjectIdentifier)
```

Comment:
okay, so you just went for it and give more errors (I don't request change here, but let's discuss this approach today).